### ISSUE TITLE
`:editor format` no longer supports onsave lsp formatting

### ISSUE THREAD

**Issue by vincowl** [doomemacs/doomemacs#7490]
### I confirm that...

- [X] I have searched the [issue tracker](https://github.com/doomemacs/doomemacs/issues), [documentation](https://docs.doomemacs.org), [FAQ](https://docs.doomemacs.org/-/faq), [Discourse](https://discourse.doomemacs.org), and [Google](https://google.com), in case this issue has already been reported/resolved.
- [X] I have read "[How to Debug Issues](https://doomemacs.org/d/how2debug)", and will use it to provide as much information about this issue as possible.
- [X] The issue can be reproduced on the **latest** available commit of Doom.
- [X] The issue can be reproduced on a stable release of Emacs, such as 27, 28, or 29. *(Unstable versions end in .50, .60, or .9x)*

### Expected behavior

When I save a buffer using a lsp server that has a format capability, doom should launch the lsp-formater through `+format/buffer`

### Current behavior

When I save a buffer using a lsp server that has a format capability, doom launches `apheleia--format-after-save` directly

### Steps to reproduce

1. Be sure to have `prettier` not installed on your system (used by apheleia)
2. Open Emacs
3. Open a badly formated html file (for instance) with html-ls installed
4. Save the file
5. The file should not be formated

### System Information

https://pastebin.com/RZQYHPaD

**Comment by vincowl** [comment]
A workaround found on the discord server could be something like : 
``` lisp
(setq-hook! 'web-mode-hook
  apheleia-inhibit t
  +format-with nil)
(add-hook 'web-mode-hook
          (lambda()
            (add-hook 'before-save-hook #'+format/buffer nil t)))
```

**Comment by fdeitylink** [comment]
I am experiencing similar issues. I use `ormolu` to format my Haskell code, which the language server uses by default. However, since `(format +onsave)` uses `apheleia-global-mode` and apheleia defaults to using `brittany`, format on save uses the incorrect formatter. Instead of
```elisp
(when (modulep! +onsave)
  (add-hook 'doom-first-file-hook #'apheleia-global-mode))
```
(written [here](https://github.com/doomemacs/doomemacs/blob/master/modules/editor/format/config.el#L34-L35)) it may be better to write
```elisp
(when (modulep! +onsave)
  (add-hook 'before-save-hook #'+format/buffer))
```
Happy to make this a PR if these seems like the right way to go. Seemed to work for me.

**Comment by goncalvesnelson** [comment]
I think the solution can be found here: https://github.com/radian-software/apheleia/discussions/120

**Comment by fdeitylink** [comment]
I believe this is fixed as of #7924 (and the refactoring work henrik did)
